### PR TITLE
Removed Mocha from app generators tests

### PR DESCRIPTION
**Pull request by ronakjangir47**
r? @kaspth

### CODE REVIEW

File: railties/test/generators/app_generator_test.rb [hunk 1]
```diff
@@ -117,35 +116,33 @@ def test_application_name_is_detected_if_it_exists_and_app_folder_renamed
 
     run_generator [app_root]
 
-    Rails.application.config.root = app_moved_root
-    Rails.application.class.stubs(:name).returns("Myapp")
-    Rails.application.stubs(:is_a?).returns(Rails::Application)
+    stub_rails_application(app_moved_root) do
+      Rails.application.stub(:is_a?, proc{ Rails::Application }) do
```
Comment (ronakjangir47) [inline railties/test/generators/app_generator_test.rb]: Without `proc` `Rails::Application` gives error "Rails::Application is abstract, you cannot instantiate it directly." I'm not sure if it's right way to do.
Also other tests are running properly without stubbing `is_a?` so removed it from other tests.

File: railties/test/generators/app_generator_test.rb [hunk 2]
```diff
@@ -651,18 +650,37 @@ def test_after_bundle_callback
...
+    check_open = -> *args do
+      assert_equal [ path, 'Accept' => 'application/x-thor-template' ], args
+      template
+    end
 
-    bundler_first = sequence('bundle, binstubs, after_bundle')
-    generator.expects(:bundle_command).with('install').once.in_sequence(bundler_first)
-    generator.expects(:bundle_command).with('exec spring binstub --all').in_sequence(bundler_first)
-    generator.expects(:run).with('echo ran after_bundle').in_sequence(bundler_first)
+    sequence = ['install', 'exec spring binstub --all', 'echo ran after_bundle']
```
Comment (ronakjangir47) [inline railties/test/generators/app_generator_test.rb]: Just want to highlight this change, is it okay?

File: railties/test/generators/app_generator_test.rb (hunk 1 above)
Comment (kaspth) [inline railties/test/generators/app_generator_test.rb]: Minitest checks sends `call` if the argument respond to it - `Rails::Application` does: https://github.com/rails/rails/blob/master/railties/lib/rails/application.rb#L161

So this is fine, but I'd prefer we use `-> { Rails::Application }` (that's with a space after `->`).

File: railties/test/generators/app_generator_test.rb [hunk 3]
```diff
@@ -538,18 +525,30 @@ def test_spring
...
   def test_spring_binstubs
     jruby_skip "spring doesn't run on JRuby"
-    generator.stubs(:bundle_command).with('install')
-    generator.expects(:bundle_command).with('exec spring binstub --all').once
-    quietly { generator.invoke_all }
+    command_check = -> command do
+      @install_called ||= 0
+
+      case command
+      when 'install'
```
Comment (kaspth) [inline railties/test/generators/app_generator_test.rb]: We need a comment explaining why we don't have to do anything about `install`.

File: railties/test/generators/app_generator_test.rb [hunk 4]
```diff
@@ -538,18 +525,30 @@ def test_spring
 
   def test_spring_binstubs
     jruby_skip "spring doesn't run on JRuby"
-    generator.stubs(:bundle_command).with('install')
-    generator.expects(:bundle_command).with('exec spring binstub --all').once
-    quietly { generator.invoke_all }
+    command_check = -> command do
+      @install_called ||= 0
```
Comment (kaspth) [inline railties/test/generators/app_generator_test.rb]: `@install_called` isn't used in the `install` case - did you forget to rename, or shouldn't we expect binstub to be called? :grin:

File: railties/test/generators/app_generator_test.rb [hunk 5]
```diff
@@ -651,18 +650,37 @@ def test_after_bundle_callback
...
-    bundler_first = sequence('bundle, binstubs, after_bundle')
-    generator.expects(:bundle_command).with('install').once.in_sequence(bundler_first)
-    generator.expects(:bundle_command).with('exec spring binstub --all').in_sequence(bundler_first)
-    generator.expects(:run).with('echo ran after_bundle').in_sequence(bundler_first)
+    sequence = ['install', 'exec spring binstub --all', 'echo ran after_bundle']
+    command_check = -> command do
+      @called ||= 0
 
-    quietly { generator.invoke_all }
+      assert sequence.include?(command) && sequence[@called] == command, "commands should be called in sequence #{sequence}"
```
Comment (kaspth) [inline railties/test/generators/app_generator_test.rb]: The `include?` doesn't make sense when we're already doing `==`. Remove it.

Comment (kaspth) [inline railties/test/generators/app_generator_test.rb]: Then just use `assert_equal`

File: railties/test/generators/app_generator_test.rb [hunk 6]
```diff
@@ -651,18 +650,37 @@ def test_after_bundle_callback
...
+      assert_equal [ path, 'Accept' => 'application/x-thor-template' ], args
+      template
+    end
 
-    bundler_first = sequence('bundle, binstubs, after_bundle')
-    generator.expects(:bundle_command).with('install').once.in_sequence(bundler_first)
-    generator.expects(:bundle_command).with('exec spring binstub --all').in_sequence(bundler_first)
-    generator.expects(:run).with('echo ran after_bundle').in_sequence(bundler_first)
+    sequence = ['install', 'exec spring binstub --all', 'echo ran after_bundle']
+    command_check = -> command do
```
Comment (kaspth) [inline railties/test/generators/app_generator_test.rb]: I think we should stick to `check` being a prefix. Sounds better to me :grin:

File: railties/test/generators/app_generator_test.rb [hunk 7]
```diff
@@ -651,18 +650,37 @@ def test_after_bundle_callback
...
+      @called ||= 0
```
Comment (kaspth) [inline railties/test/generators/app_generator_test.rb]: `@called` feels off, how about `@sequence_step`?

File: railties/test/generators/app_generator_test.rb (hunk 6 above)
Comment (kaspth) [inline railties/test/generators/app_generator_test.rb]: But in this case it isn't a command check. It's `ensure_bundler_first` :wink:

**Comment by kaspth** [comment]
@ronakjangir47 seems like there were some copy and paste scuffs. Even though these removals are pretty routine, please think about what you call variables and such. Thanks! :smile:

**Comment by ronakjangir47** [comment]
@kaspth Thanks for your feedback and Sorry for naming conventions :wink: 
Updated PR, please have a look again.

**Comment by kaspth** [comment]
Thanks!

**Pull request merged**